### ISSUE TITLE
Social Card 4: Sharing a members category of votes. eg. Voted consistently for.

### ISSUE THREAD

**Issue by Taz17** [openaustralia/theyvoteforyou#1371]
Social card 4 is for when a user shares what a member consistently votes for (including what cannot be decided on)

The categories can be found on a members page. e.g. https://theyvoteforyou.org.au/people/senate/tasmania/carol_brown

@tseyiokorodudu and I discussed what this card would look like and how to rank the policies shown on the card.

Here is a rough design of what it would look like:

![image](https://user-images.githubusercontent.com/86769131/158299622-901cc529-d38c-4f1c-8eb0-8b95a26264d6.png)

The `+ 20 Others` in the bottom right hand corner is to show how many more policies fall under the shared category.

As for the policies shown on the the card I suggested we display the most subscribed to policies as a higher subscriber count tells us how interesting end users find a policy

**Comment by Taz17** [comment]
@mlandauer @kat I'd love to know what you think about how to choose what policies show up on the card.

A known downside of using the subscriber count is that it would generally show the same policies for members of the same party which have been in parliament for quite a time. This might create a separation between parties

It also would limit which policies get displayed since its likely to pick the same policies over and over again.

**Comment by mlandauer** [comment]
Thanks @Taz17 @tseyiokorodudu!

I like the continuing use of the "+20 others" pattern. Feels like a very solid (but non-intrusive) way to hint at all the goodies that await if someone follows the links. That's great.

Is there a reason for not including a picture of the member here?

I'm not sure about the "white line" separators being used here. They're not used on the site and the site uses whitespace (and careful choice of line spacing) to keep things separated.

As for the order of the policies. I agree that there are problems with using the subscriber count. Intuitively I don't think there's going to be an "ideal" way here. All the policies are not created equal. Some are going to be super important to some people, etc, etc. So, my inclination would be to just mirror what the site does. It's obviously problematic but it's the problematic that we have now. They're just sorted alphabetically. This is so obviously arbitrary.

**Comment by tseyiokorodudu** [comment]
There's no particular reason the image of the member isn't included. We can definitely add it. 

White was just a temporary colour choice for the line while we figure out a better color. In using the line as a separator, we were trying to stay in line with the design the website uses for this particular content. I will try out the white space and see how it looks.

This is noted
> So, my inclination would be to just mirror what the site does. It's obviously problematic but it's the problematic that we have now. They're just sorted alphabetically. This is so obviously arbitrary.

**Comment by tseyiokorodudu** [comment]
What do we think about this?

![image](https://user-images.githubusercontent.com/31440427/159221200-b5759145-2b20-4a49-b11a-ec6ace0be679.png)

Note: The text on the card is hard-coded.

**Comment by Taz17** [comment]
Oh a new category, thats nice! I really like this design, and much prefer it that the previous design. But don't you think the image would be too small when twitter/facebook shrinks them down?

**Comment by tseyiokorodudu** [comment]
Yeah, it will be small but still recognisable. It's the same image size as those on the policy card, i.e. `large_image_size`. So you could look at that to see how it will turn out.

**Comment by mlandauer** [comment]
I've merged the three PR's for social card 4, made a few small changes, and rolled it out to production.

I haven't run the rake task to generate the images as I don't want to put any extra load on the server right now while we have increased traffic today because of the SMH article.

Looking at the design of this card now in a little more detail I think there are a few things that I think could be improved.

The different left indentation for the heading "Eric Abetz voted a mixture" versus the list below makes it not super obvious to me that one flows on from the other. Is that just me? @katska what do you think?

Also I noticed in production when there are only a small number of policies listed the formatting gets a little confused. See for instance https://theyvoteforyou.org.au/people/representatives/bass/bridget_archer?card=true&category=for2

Also the "+20 Others" doesn't make sense to me on the right hand side here. The list is left aligned. Your eye scans down the list on the left hand side. It's kind of easy to miss the "+20 others" on the right hand side.

**Comment by Taz17** [comment]
@mlandauer I think the "+ 20 Others" is fine when the list is taking up the whole card.

`Find out more: theyvoteforyou.org.au/people/representatives/..../................`  Takes space and moving the "+ 20 Others" below the list would mean we would have to change the position of:
- `Find out more: theyvoteforyou.org.au/people/representatives/..../................`

**Comment by katska** [comment]
"The different left indentation for the heading "Eric Abetz voted a mixture" versus the list below makes it not super obvious to me that one flows on from the other. Is that just me? @katska what do you think?"
It wasn't obvious to me, so I think they need to be indented or otherwise made more obviously a list. Does that make sense @tseyiokorodudu  @Taz17

**Comment by tseyiokorodudu** [comment]
The challenge I had with the design of this card, especially with the addition of the MP's image, was maintaining a balance between the space and choosing the right font size, so the contents fit perfectly within the card. 

So moving "+20 Others" to the left-hand side would mean having to reduce the font size of the policies (which I fear might make them too small on Twitter & Facebook) or reducing the white space between the policies (which will make the card look crammed).

**Comment by tseyiokorodudu** [comment]
> so I think they need to be indented or otherwise made more obviously a list. Does that make sense @tseyiokorodudu @Taz17

@katska Yes, it does. What instantly comes to mind is maybe adding a bullet point? Is that what you mean?

**Comment by tseyiokorodudu** [comment]
> The challenge I had with the design of this card, especially with the addition of the MP's image, was maintaining a balance between the space and choosing the right font size, so the contents fit perfectly within the card.
> 
> So moving "+20 Others" to the left-hand side would mean having to reduce the font size of the policies (which I fear might make them too small on Twitter & Facebook) or reducing the white space between the policies (which will make the card look crammed).

Alternatively, we could reduce the number of policies on the card from 4 to 3.

**Comment by tseyiokorodudu** [comment]
Moved the "20 others" to the left and reduced the number of policies to 3....Here is the result

![image](https://user-images.githubusercontent.com/31440427/159806513-916a627a-c2cc-4635-9227-a95d84aebad6.png)

> so I think they need to be indented or otherwise made more obviously a list

Regarding the indentation, this is what it looks like when we indent the policies. I don't think it looks good. 

![image](https://user-images.githubusercontent.com/31440427/159806911-7e32c122-620e-4b12-985a-e57b0aef5eb8.png)

Alternatively, we could revert to using the lines

![image](https://user-images.githubusercontent.com/31440427/159806737-cfeec478-cfe1-46bc-a036-39f5116d8df3.png)

@mlandauer @katska Thoughts?

**Comment by katska** [comment]
Hi @tseyiokorodudu 
I think the lines are ok. 
As you'd mentioned the bullet point thing I had a quick go at mocking that up visually, along with rebalancing the other elements to go with that 
- bigger photo
-  even spacing around photo
- Headline aligned with top of photo
- slightly smaller text
- just for fun I also added an extra small border beneath the main banner to soften blend between the two parts. Maybe there could be a bit more space again underneath that now.   
<img width="753" alt="Screen Shot 2022-03-24 at 2 46 20 pm" src="https://user-images.githubusercontent.com/1001129/159838214-4dfea98a-ff46-4aaf-86ca-a0ef6df6ba3d.png">

**Comment by katska** [comment]
With a bit more space added back in - I think that's more balanced again
<img width="748" alt="Screen Shot 2022-03-24 at 3 59 07 pm" src="https://user-images.githubusercontent.com/1001129/159845485-d42907b3-0284-448c-84c1-b18ec8cee224.png">

**Comment by katska** [comment]
A couple of variations - what happens when the arrangement is tweaked to have a bit more indent, like a traditional bulleted list, and how important is the relationship between the headline and options shown? 

<img width="752" alt="Screen Shot 2022-03-24 at 5 49 36 pm" src="https://user-images.githubusercontent.com/1001129/159858805-986e12b7-52f1-4528-9204-94f011ba26e9.png">

<img width="750" alt="Screen Shot 2022-03-24 at 5 52 18 pm" src="https://user-images.githubusercontent.com/1001129/159859179-1c7540a1-e0a3-4aec-8fa5-b88df1e03bda.png">

I know there will be lots of other variations that work, and I'm tinkering with the fine details of what brings one version that you're working on together.

**Comment by tseyiokorodudu** [comment]
Thanks, @katska for the mockup. It definitely looks more spacious.

**Comment by tseyiokorodudu** [comment]
Here is the card based on @katska mock-up:

![image](https://user-images.githubusercontent.com/31440427/160315653-63835716-e10d-4c77-8b5e-e8f68e8df5c4.png)

Still working out what happens when we have just one policy but so far we've:
- Made sure the URL stays below
- And removed "+ 20 others" 

![image](https://user-images.githubusercontent.com/31440427/160316998-fb989680-06c1-4f76-822d-c94425a0e740.png)

**Comment by katska** [comment]
@tseyiokorodudu I feel like that's getting there. How does this direction seem to you, is there anything that doesn't work. Do you have other suggestions we're not considering?
If going along this path then - bigger photo yes. I think aligning the bulleted issues as you have done is good, so probably doesn't need further indenting. I wasn't sure about that when I added it. I'd appreciate your thoughts on that too. Also if these particular bullets are the best option?

**Comment by katska** [comment]
@tseyiokorodudu hey I notice there still seems to be a bit less space above the photo than to the left. The idea was an even border. Ideally this and the other layout features scale for different screens/devices.

**Comment by tseyiokorodudu** [comment]
Hi @katska, I like the direction; it isn't so different from the original and it solves the minor issues with the original. 

About the bullet points, my thought on it constantly changes, sometimes I look at it and I think it works and other times, I don't think it does. However, Iooking at it from a purely functional perspective; I think it does the work of showing that it is a list and that the policies are linked to the title heading. Based on that, I think we could leave the bullet points in. 

> Also if these particular bullets are the best option?

We could definitely try different bullet types and see which one looks best. 

Balancing the space and content has been the major struggle with this card. However, Mutaz and I will talk it through and figure out the best course of action.

**Comment by Taz17** [comment]
@katska, Tseyi and I have worked on the card and we updated the spacing around the image to be all the same (left, right, and top spacing).

We are now working on shrinking the font size when the policy names are too long and cause the text on the car to overflow.

**Comment by katska** [comment]
>Balancing the space and content has been the major struggle with this card. However, Mutaz and I will talk it through and figure out the best course of action.
I'm curious to see your thinking on this as you make decisions to better understand where the challenges are/have been. 

Style of bullets is a minor point and shouldn't be a blocker. I just did the default thing as a functional starting point. It needs to feel simple and not get in the way at the same time and "showing that it is a list and that the policies are linked to the title heading". Lets leave that one unless there's time at the end when we've ironed out the whole flow 
:) Feels so close now!

**Comment by Taz17** [comment]
Here is what the card currently looks like. We fixed up the spacing around the image. Now to the left, right, and top of the image there is a 60 pixel gap. We also added the ability to shrink the size of the policy names. 

Before shrinking:
![image](https://user-images.githubusercontent.com/86769131/160738692-8ef7d35d-f071-4f47-a9a2-a06ae2229969.png)

After shrinking:
![image](https://user-images.githubusercontent.com/86769131/160738519-73dc3e9e-a26b-45cb-bef7-12e02d434171.png)

@katska, and @mlandauer we'd love your thoughts about this. We are aware that the spacing between the "+ 30 Others" and the URL is small and thats because it is being shrunk until the two no longer overlap. We tried fixing that but the result will always depend on the amount of text there is to begin with. Meaning we cant always get what we want.
